src/session_impl.cpp

```diff
@@ -5516,6 +5519,14 @@ namespace aux {
 		m_dht_settings = settings;
 	}
 
+	bool session_impl::set_dht_storage(dht::dht_storage_constructor_type sc)
+	{
+		if (m_dht) return false;
```

Comment:
This may be a bit surprising semantics. In a sense, is it important that the DHT is shut down first, before setting this?
another way of looking at it is that this setting takes effect at the next restart of the DHT, and you can still set it first and then restart.